### PR TITLE
bwidget 1.9.16 (new formula)

### PR DESCRIPTION
**Pull request by willcohen**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by willcohen** [comment]
This hits one error with the new formula audit. However, given that these files are meant to go into TCL_LIBRARY_PATH (see https://github.com/tcltk/bwidget), this seems like the appropriate location.
```
bwidget:
  * Non-libraries were installed to "/opt/homebrew/opt/bwidget/lib".
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /opt/homebrew/opt/bwidget/lib/bitmap.tcl
      /opt/homebrew/opt/bwidget/lib/passwddlg.tcl
      /opt/homebrew/opt/bwidget/lib/separator.tcl
      /opt/homebrew/opt/bwidget/lib/button.tcl
      /opt/homebrew/opt/bwidget/lib/panedw.tcl
      /opt/homebrew/opt/bwidget/lib/utils.tcl
      /opt/homebrew/opt/bwidget/lib/widget.tcl
      /opt/homebrew/opt/bwidget/lib/ChangeLog
      /opt/homebrew/opt/bwidget/lib/tree.tcl
      /opt/homebrew/opt/bwidget/lib/titleframe.tcl
      /opt/homebrew/opt/bwidget/lib/spinbox.tcl
      /opt/homebrew/opt/bwidget/lib/color.tcl
      /opt/homebrew/opt/bwidget/lib/messagedlg.tcl
      /opt/homebrew/opt/bwidget/lib/dragsite.tcl
      /opt/homebrew/opt/bwidget/lib/statusbar.tcl
      /opt/homebrew/opt/bwidget/lib/mainframe.tcl
      /opt/homebrew/opt/bwidget/lib/xpm2image.tcl
      /opt/homebrew/opt/bwidget/lib/entry.tcl
      /opt/homebrew/opt/bwidget/lib/notebook.tcl
      /opt/homebrew/opt/bwidget/lib/buttonbox.tcl
      /opt/homebrew/opt/bwidget/lib/font.tcl
      /opt/homebrew/opt/bwidget/lib/pagesmgr.tcl
      /opt/homebrew/opt/bwidget/lib/dialog.tcl
      /opt/homebrew/opt/bwidget/lib/labelentry.tcl
      /opt/homebrew/opt/bwidget/lib/arrow.tcl
      /opt/homebrew/opt/bwidget/lib/CHANGES.txt
      /opt/homebrew/opt/bwidget/lib/pkgIndex.tcl
      /opt/homebrew/opt/bwidget/lib/scrollframe.tcl
      /opt/homebrew/opt/bwidget/lib/label.tcl
      /opt/homebrew/opt/bwidget/lib/progressdlg.tcl
      /opt/homebrew/opt/bwidget/lib/panelframe.tcl
      /opt/homebrew/opt/bwidget/lib/wizard.tcl
      /opt/homebrew/opt/bwidget/lib/progressbar.tcl
      /opt/homebrew/opt/bwidget/lib/combobox.tcl
      /opt/homebrew/opt/bwidget/lib/README.txt
      /opt/homebrew/opt/bwidget/lib/scrollview.tcl
      /opt/homebrew/opt/bwidget/lib/labelframe.tcl
      /opt/homebrew/opt/bwidget/lib/LICENSE.txt
      /opt/homebrew/opt/bwidget/lib/listbox.tcl
      /opt/homebrew/opt/bwidget/lib/scrollw.tcl
      /opt/homebrew/opt/bwidget/lib/dynhelp.tcl
      /opt/homebrew/opt/bwidget/lib/dropsite.tcl
      /opt/homebrew/opt/bwidget/lib/init.tcl
Error: 1 problem in 1 formula detected
```

**Comment by willcohen** [comment]
@gromgit @SMillerDev #113812 is the only other mention I can find about the TCL_LIBRARY_PATH. If there's a preferred workaround to copying the files to `lib` for this TCL library, I'd be happy to do that instead.

**Comment by willcohen** [comment]
Additionally, as context, copying to lib is the method nixpkgs uses for the same package: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/bwidget/default.nix

**Comment by willcohen** [comment]
Latest force push adds `/share` to the `TCL_LIBRARY_PATH` in the formula for `tcl-tk`, allowing `bwidget` to install to `/share` instead of `/lib` and still have its files found by tclsh.

**Comment by willcohen** [comment]
All remaining test failures appear to be from flaky pulls from SF. I think this is ready for review.

**Comment by gromgit** [comment]
> @gromgit @SMillerDev #113812 is the only other mention I can find about the TCL_LIBRARY_PATH. If there's a preferred workaround to copying the files to `lib` for this TCL library, I'd be happy to do that instead.

Please don't change the `tcl-tk` formula. All other formulae installing Tcl modules should install them under their own `#{lib}`, then Homebrew will take care of symlinking all the files into the correct place.

File: Formula/bwidget.rb
Comment (gromgit) [inline Formula/bwidget.rb:11]:
```suggestion
    lib/"bwidget".install Dir["*"]
```

File: Formula/bwidget.rb
Comment (gromgit) [inline Formula/bwidget.rb:8]:
```suggestion
  depends_on "tcl-tk"
```
The stock macOS Tcl doesn't know about Homebrew anyway.

**Review by gromgit — CHANGES_REQUESTED** [review]
Don't change the `tcl-tk` formula.

File: Formula/bwidget.rb
Comment (willcohen) [reply Formula/bwidget.rb:11]: Apologies for the syntax question. This fails with `NoMethodError: undefined method `install' for "bwidget":String`.

There's no actual installer here -- the method as I understand it is just to copy all the *.tcl files to lib, and the lang and images directories over subdirectories of lib.

I tried using the following, but then the images are simply copied to the top-level and the lang appears to have nothing happen. Per https://ruby-doc.org/stdlib-2.4.1/libdoc/fileutils/rdoc/FileUtils.html#method-c-cp_r I assume this should be working but I'm clearly doing this wrong.
```
cp_r Dir.glob("*.tcl"), "#{lib}"
cp_r "images", "#{lib}"
cp_r "lang", "#{lib}"
```

As seen above, `cp_r ".", "#{lib}"` does work without modifying `tcl-tk`, with the unpleasant side effect of adding the readme and changelog.

**Comment by willcohen** [comment]
Latest force push fixes commit style as suggested by CI. CI now fully passes.

> Don't change the `tcl-tk` formula.

Understood -- was taking the audit for new formula too literally in an attempt to get the green check mark. Once I resolve how to copy the *.tcl files without a complete `cp_r "."` will re-push with just the one formula.

File: Formula/bwidget.rb
Comment (gromgit) [reply Formula/bwidget.rb:11]: > Apologies for the syntax question. This fails with `NoMethodError: undefined method 'install' for "bwidget":String`.

Sorry, I left out a pair of parentheses. That should've been:
```
    (lib/"bwidget").install Dir["*"]
```

> There's no actual installer here -- the method as I understand it is just to copy all the *.tcl files to lib, and the lang and images directories over subdirectories of lib.

That's what the `install` method on the `lib/"widget"` pathname does.

File: Formula/bwidget.rb
Comment (willcohen) [reply Formula/bwidget.rb]: Change made with 7323f19.

File: Formula/bwidget.rb
Comment (willcohen) [reply Formula/bwidget.rb]: Change made with 7323f19.

**Comment by willcohen** [comment]
@gromgit Many thanks.

Latest force push incorporates the changes.
`brew test bwidget` passes on macOS locally.
`brew audit --strict bwidget` passes on macOS locally.
`brew audit --new bwidget` fails locally with `bwidget: * Dependency 'tcl-tk' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.`

**Comment by willcohen** [comment]
Given the keg-only issue, does this package need a caveat added that this library will only be found if tcl-tk is found first in the path?

**Comment by gromgit** [comment]
> `Dependency 'tcl-tk' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.`

Ah, that's a bit of a bummer. Both this and `expect` are equivalent formulae in that both provision Tcl modules, so both should have the same dependency requirement viz. Tcl. `expect` just doesn't trigger this check because it's not a new formula.

@Homebrew/core, any guidance on how to resolve this? `uses_on_macos` is pointless because the system Tcl simply won't see this module, so tests on macOS would either always fail or (as with `expect` previously) end up loading and testing the similarly-named system Tcl module instead.

**Comment by willcohen** [comment]
I should note that in addition to this, as currently written, CI is going to fail (as seen on this previous run, failure pasted for posterity. See https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:92). This was why I'd previously modified it to copy to `/share` instead of `/lib`, and my read of https://github.com/Homebrew/brew/issues/2008 suggested that since `.tcl` are not likely to be an exception to this rule, modifying `TCL_LIBRARY_PATH` to include `share` was the way to go.
```
==> brew audit bwidget --online --new-formula
[26](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:27)
==> FAILED
[27](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:28)
Full audit bwidget --online --new-formula output
[28](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:29)
  bwidget:
[29](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:30)
    * Non-libraries were installed to "/opt/homebrew/opt/bwidget/lib".
[30](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:31)
      Installing non-libraries to "lib" is discouraged.
[31](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:32)
      The offending files are:
[32](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:33)
        /opt/homebrew/opt/bwidget/lib/bitmap.tcl
[33](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:34)
        /opt/homebrew/opt/bwidget/lib/passwddlg.tcl
[34](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:35)
        /opt/homebrew/opt/bwidget/lib/separator.tcl
[35](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:36)
        /opt/homebrew/opt/bwidget/lib/button.tcl
[36](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:37)
        /opt/homebrew/opt/bwidget/lib/panedw.tcl
[37](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:38)
        /opt/homebrew/opt/bwidget/lib/utils.tcl
[38](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:39)
        /opt/homebrew/opt/bwidget/lib/widget.tcl
[39](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:40)
        /opt/homebrew/opt/bwidget/lib/ChangeLog
[40](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:41)
        /opt/homebrew/opt/bwidget/lib/tree.tcl
[41](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:42)
        /opt/homebrew/opt/bwidget/lib/titleframe.tcl
[42](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:43)
        /opt/homebrew/opt/bwidget/lib/spinbox.tcl
[43](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:44)
        /opt/homebrew/opt/bwidget/lib/color.tcl
[44](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:45)
        /opt/homebrew/opt/bwidget/lib/messagedlg.tcl
[45](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:46)
        /opt/homebrew/opt/bwidget/lib/dragsite.tcl
[46](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:47)
        /opt/homebrew/opt/bwidget/lib/statusbar.tcl
[47](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:48)
        /opt/homebrew/opt/bwidget/lib/mainframe.tcl
[48](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:49)
        /opt/homebrew/opt/bwidget/lib/xpm2image.tcl
[49](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:50)
        /opt/homebrew/opt/bwidget/lib/entry.tcl
[50](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:51)
        /opt/homebrew/opt/bwidget/lib/notebook.tcl
[51](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:52)
        /opt/homebrew/opt/bwidget/lib/buttonbox.tcl
[52](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:53)
        /opt/homebrew/opt/bwidget/lib/font.tcl
[53](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:54)
        /opt/homebrew/opt/bwidget/lib/pagesmgr.tcl
[54](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:55)
        /opt/homebrew/opt/bwidget/lib/dialog.tcl
[55](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:56)
        /opt/homebrew/opt/bwidget/lib/labelentry.tcl
[56](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:57)
        /opt/homebrew/opt/bwidget/lib/arrow.tcl
[57](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:58)
        /opt/homebrew/opt/bwidget/lib/CHANGES.txt
[58](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:59)
        /opt/homebrew/opt/bwidget/lib/pkgIndex.tcl
[59](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:60)
        /opt/homebrew/opt/bwidget/lib/scrollframe.tcl
[60](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:61)
        /opt/homebrew/opt/bwidget/lib/label.tcl
[61](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:62)
        /opt/homebrew/opt/bwidget/lib/progressdlg.tcl
[62](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:63)
        /opt/homebrew/opt/bwidget/lib/panelframe.tcl
[63](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:64)
        /opt/homebrew/opt/bwidget/lib/wizard.tcl
[64](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:65)
        /opt/homebrew/opt/bwidget/lib/progressbar.tcl
[65](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:66)
        /opt/homebrew/opt/bwidget/lib/combobox.tcl
[66](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:67)
        /opt/homebrew/opt/bwidget/lib/README.txt
[67](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:68)
        /opt/homebrew/opt/bwidget/lib/scrollview.tcl
[68](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:69)
        /opt/homebrew/opt/bwidget/lib/labelframe.tcl
[69](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:70)
        /opt/homebrew/opt/bwidget/lib/LICENSE.txt
[70](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:71)
        /opt/homebrew/opt/bwidget/lib/listbox.tcl
[71](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:72)
        /opt/homebrew/opt/bwidget/lib/scrollw.tcl
[72](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:73)
        /opt/homebrew/opt/bwidget/lib/dynhelp.tcl
[73](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:74)
        /opt/homebrew/opt/bwidget/lib/dropsite.tcl
[74](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:75)
        /opt/homebrew/opt/bwidget/lib/init.tcl
[75](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:76)
  Error: 1 problem in 1 formula detected
[76](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:78)
==> brew bottle --verbose --json bwidget --only-json-tab
[77](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:79)
==> brew bottle --merge --write --no-commit --no-all-checks ./bwidget--1.9.16.arm64_monterey.bottle.json
[78](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:80)
==> brew uninstall --force bwidget
[79](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:81)
==> brew install --only-dependencies ./bwidget--1.9.16.arm64_monterey.bottle.tar.gz
[80](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:82)
==> brew install ./bwidget--1.9.16.arm64_monterey.bottle.tar.gz
[81](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:83)
==> brew linkage --test bwidget
[82](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:84)
==> brew linkage --cached --test --strict bwidget
[83](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:85)
==> brew linkage --cached bwidget
[84](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:86)
==> brew install --only-dependencies --include-test bwidget
[85](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:87)
==> brew test --verbose bwidget
[86](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:88)

[87](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:89)
Warning: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
[88](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:90)
Error: 1 failed step!
[89](https://github.com/Homebrew/homebrew-core/actions/runs/3942007982/jobs/6745243337#step:6:91)
brew audit bwidget --online --new-formula
```

**Comment by willcohen** [comment]
@gromgit is xorgproto widely used? If tcl-tk is marked as just conflicting with xorgproto does that allow tcl-tk to be non-keg-only?

**Comment by gromgit** [comment]
> as currently written, CI is going to fail one more time (as seen on this previous run, failure pasted for posterity

It shouldn't. I just pulled down your PR onto my M1 Mac mini, ran `brew install -s bwidget`, and:
```
$ find /opt/homebrew/opt/bwidget/lib/ -type f | head -5
/opt/homebrew/opt/bwidget/lib//bwidget/demo/basic.tcl
/opt/homebrew/opt/bwidget/lib//bwidget/demo/select.tcl
/opt/homebrew/opt/bwidget/lib//bwidget/demo/tree.tcl
/opt/homebrew/opt/bwidget/lib//bwidget/demo/manager.tcl
/opt/homebrew/opt/bwidget/lib//bwidget/demo/tmpldlg.tcl

$ find /opt/homebrew/lib/bwidget/ -type f | head -5
/opt/homebrew/lib/bwidget//demo/basic.tcl
/opt/homebrew/lib/bwidget//demo/select.tcl
/opt/homebrew/lib/bwidget//demo/tree.tcl
/opt/homebrew/lib/bwidget//demo/manager.tcl
/opt/homebrew/lib/bwidget//demo/tmpldlg.tcl
```
Clearly, these aren't the paths the CI run you pasted are complaining about. After changing `depends_on` to `uses_from_macos` to silence that particular audit complaint:
```
$ brew audit --online --new-formula bwidget

```
And to  prove `brew audit` didn't fall asleep at the wheel:
```
$ touch /opt/homebrew/opt/bwidget/lib/test.tcl
$ brew audit --online --new-formula bwidget
bwidget:
  * Non-libraries were installed to "/opt/homebrew/opt/bwidget/lib".
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /opt/homebrew/opt/bwidget/lib/test.tcl
Error: 1 problem in 1 formula detected
```
So your fears are quite premature. At this point, the only issue that seems to be outstanding is the type of `tcl-tk` dependency for macOS.

**Comment by willcohen** [comment]
> > as currently written, CI is going to fail one more time (as seen on this previous run, failure pasted for posterity
> 
> It shouldn't. I just pulled down your PR onto my M1 Mac mini, ran `brew install -s bwidget`, and:
> 
> ```
> $ find /opt/homebrew/opt/bwidget/lib/ -type f | head -5
> /opt/homebrew/opt/bwidget/lib//bwidget/demo/basic.tcl
> /opt/homebrew/opt/bwidget/lib//bwidget/demo/select.tcl
> /opt/homebrew/opt/bwidget/lib//bwidget/demo/tree.tcl
> /opt/homebrew/opt/bwidget/lib//bwidget/demo/manager.tcl
> /opt/homebrew/opt/bwidget/lib//bwidget/demo/tmpldlg.tcl
> 
> $ find /opt/homebrew/lib/bwidget/ -type f | head -5
> /opt/homebrew/lib/bwidget//demo/basic.tcl
> /opt/homebrew/lib/bwidget//demo/select.tcl
> /opt/homebrew/lib/bwidget//demo/tree.tcl
> /opt/homebrew/lib/bwidget//demo/manager.tcl
> /opt/homebrew/lib/bwidget//demo/tmpldlg.tcl
> ```
> 
> Clearly, these aren't the paths the CI run you pasted are complaining about. After changing `depends_on` to `uses_from_macos` to silence that particular audit complaint:
> 
> ```
> $ brew audit --online --new-formula bwidget
> ```
> 
> And to prove `brew audit` didn't fall asleep at the wheel:
> 
> ```
> $ touch /opt/homebrew/opt/bwidget/lib/test.tcl
> $ brew audit --online --new-formula bwidget
> bwidget:
>   * Non-libraries were installed to "/opt/homebrew/opt/bwidget/lib".
>     Installing non-libraries to "lib" is discouraged.
>     The offending files are:
>       /opt/homebrew/opt/bwidget/lib/test.tcl
> Error: 1 problem in 1 formula detected
> ```
> 
> So your fears are quite premature. At this point, the only issue that seems to be outstanding is the type of `tcl-tk` dependency for macOS.

Right. Another reason why my `cp_r` was the wrong move before!

**Comment by gromgit** [comment]
> @gromgit is xorgproto widely used? If tcl-tk is marked as just conflicting with xorgproto does that allow tcl-tk to be non-keg-only?

Nope, there are formulae that depend on _both_:
```
$ brew uses --recursive --eval-all --include-build --include-test --formula tcl-tk xorgproto
arxiv_latex_cleaner       blastem                   liquidctl                 r                         weasyprint
aws-google-auth           fdroidserver              mapproxy                  textract                  woob
bazarr                    gmsh                      ocrmypdf                  torchvision               wxpython
binwalk                   graph-tool                pillow                    trezor-agent
```

**Comment by willcohen** [comment]
Was just thinking about this some more -- while the uselessness is clearly true, `uses_on_macos` seems like a useful signal to future developers that something is funny here as long as `tcl-tk` has to stay keg-only. Perhaps the expedient path is to do `uses_on_macos` (and then on OS.linux? to use `depends`), and add a caveat to double-indicate to users rather than developers that they'll need to add homebrew tclsh to the path for this to work?

Any explanation of how to use this for downstream Mac users will require them navigating this tclsh/keg issue too, so it seems like two signals rather than one is the best way to avoid this causing more questions (... for that untold number of Tcl users still out there...)

**Comment by willcohen** [comment]
@carlocab @gromgit Any thoughts on how to proceed with this PR?

**Comment by willcohen** [comment]
@gromgit Looks like the audit failure can be ignored: https://github.com/Homebrew/brew/issues/14405#issuecomment-1400680213

**Comment by gromgit** [comment]
Since that's the only CI test failure, I think it's good to do, but as I'm not a maintainer, someone else from @Homebrew/core would have to approve it.

**Comment by iMichka** [comment]
Why does it need `tcl-tk` for the build, as you seem to only install some scripts.
Would `depends_on "tcl-tk" => :test` work?

**Comment by willcohen** [comment]
It's possible I am misunderstanding how strictly homebrew thinks about dependencies. BWidget is one component of the tcllib project of standard routines for tcl. Many Tcl distributions include these libraries in addition to the main tcl interpreter. While the install could complete, without a tcl interpreter present, these routines have no value to end users -- I assumed that it makes more sense for anyone installing bwidget to have tcl-tk come along for the ride too, especially since the only tcl interpreter this is configured to work with is the homebrew one.

Is that incorrect, and should I take the dependency out?

**Comment by willcohen** [comment]
For example, on [debian](https://packages.debian.org/bullseye/bwidget), bwidget has `tk` as a dependency, which itself has `tcl` as another dependency. [nixpkgs](https://github.com/NixOS/nixpkgs/blob/216fbe09999764469066b1ccf05dcff45cf078e7/pkgs/development/libraries/bwidget/default.nix#L1) similarly requires both tcl and tk for the installation to be considered complete.

**Comment by willcohen** [comment]
(Another option would have been to stuff it into `tcl-tk` too with all the other [currently bundled resources](https://github.com/Homebrew/homebrew-core/blob/master/Formula/tcl-tk.rb#L40-L64), but given that users of that existing formula do fine without bwidget, it seemed more sensible for me to break it out entirely. `tcllib`, for example, is already bundled inside `tcl-tk`, though it packages `bwidget` separately, which is why this is still needed.

**Comment by gromgit** [comment]
> Why does it need `tcl-tk` for the build, as you seem to only install some scripts. Would `depends_on "tcl-tk" => :test` work?

BWidget is a Tk module, so it really needs a Tcl runtime at, well, run time, not just build or test time.

**Comment by iMichka** [comment]
This should be `depends_on "tcl-tk"` then, to make it a full runtime dependency.

**Comment by willcohen** [comment]
@iMichka thanks. Change made with push to 42862e0.

**Comment by willcohen** [comment]
@iMichka With the dependency issue addressed, do you see any other issues with this formula?

**Review by iMichka — DISMISSED** [review]
Looks good to me, maybe we can get one more review by another maintainer before merging

File: Formula/bwidget.rb
Comment (gromgit) [inline Formula/bwidget.rb:28]:
```suggestion
    test_bwidget = <<~EOS
      puts [package require BWidget]
      exit
    EOS
    assert_equal version.to_s, pipe_output("#{Formula["tcl-tk"].bin}/tclsh", test_bwidget).chomp
```
The original test runs the system `tclsh` and tests the system BWidget. This becomes clear when you compare the package version returned by `package require` (1.9.1 for system BWidget), so it's worth checking that instead of a simple `OK`.

Fixing that also means there's no reason to limit testing to Monterey and above.

**Review by gromgit — CHANGES_REQUESTED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by willcohen** [comment]
Apologies for losing track of the test submission change!

**Comment by chenrui333** [comment]
> Apologies for losing track of the test submission change!

No worries, thanks for your contributions!!

**Pull request closed without merging**
